commodore/inventory/parameters.py

```diff
@@ -0,0 +1,229 @@
+        self._distributions = self._find_values(DefaultsFact.DISTRIBUTION)
+        self._clouds = self._find_values(DefaultsFact.CLOUD)
+        self._cloud_regions: Dict[str, Iterable[str]] = {}
+        for cloud in self._clouds:
+            r = self._find_values(DefaultsFact.REGION, cloud=cloud)
+            self._cloud_regions[cloud] = [it for it in r if it != "params"]
+
+    @property
+    def directory(self) -> Path:
+        return self._inventory.inventory_dir
+
+    @property
+    def classes_dir(self) -> Path:
+        return self._inventory.classes_dir
+
+    @property
+    def targets_dir(self) -> Path:
+        return self._inventory.targets_dir
+
+    @property
+    def global_dir(self) -> Path:
+        return self._global_dir
+
+    def _reclass_config(self) -> Dict:
+        return {
+            "storage_type": "yaml_fs",
+            "inventory_base_uri": str(self.directory.absolute()),
+            "nodes_uri": str(self.targets_dir.absolute()),
+            "classes_uri": str(self.classes_dir.absolute()),
+            "compose_node_name": False,
+            "allow_none_override": True,
+            "ignore_class_notfound": True,
+        }
+
+    def reclass(self, invfacts: InventoryFacts) -> InventoryParameters:
+        distribution = invfacts.distribution
+        cloud = invfacts.cloud
+        region = invfacts.region
+        if distribution is None:
+            distribution = "x-fake-distribution"
+        if cloud is None:
+            cloud = "x-fake-cloud"
+        if region is None:
+            region = "x-fake-region"
+
+        c: Dict[str, Any] = {
+            "id": "c-bar",
+            "tenant": "t-foo",
+            "displayName": "Foo Inc. Bar cluster",
+            "facts": {
+                "distribution": distribution,
+                "cloud": cloud,
+                "lieutenant-instance": "lieutenant-prod",
+                f"{distribution}_version": "1.20",
+            },
+            "gitRepo": {
+                "url": "not-a-real-repo",
+            },
+        }
+        if region:
+            c["facts"]["region"] = region
+
+        cluster = Cluster(
+            cluster_response=c,
+            tenant_response={
+                "id": "t-foo",
+                "displayName": "Foo Inc.",
+                "gitRepo": {
+                    "url": "not-a-real-repo",
+                },
+            },
+        )
+        params = render_params(self._inventory, cluster)
+        # don't support legacy component_versions key
+        del params["parameters"]["components"]
+        del params["parameters"]["component_versions"]
+        params["parameters"]["openshift"] = {
+            "infraID": "infra-id",
+            "clusterID": "cluster-id",
+            "registryBucket": "x-fake-registry-bucket",
+        }
+        params["parameters"]["cloud"]["availabilityZones"] = []
+        params["parameters"]["cloud"]["projectName"] = "x-fake-project"
```

Comment:
I guess these are the VSHN specific parameter you mentioned?

Reply:
Refactored the command in the latest version to not need those anymore.